### ISSUE TITLE
Settings_MaxConcurrentStreamsPost_Server test failure

### ISSUE THREAD

**Issue by BrennanConroy** [dotnet/aspnetcore#27370]
```
System.Threading.Tasks.TaskCanceledException : A task was canceled.
   at System.Net.Http.Http2Connection.SendStreamDataAsync(Int32 streamId, ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)
   at System.Net.Http.Http2Connection.Http2Stream.SendDataAsync(ReadOnlyMemory`1 buffer, CancellationToken cancellationToken)
   at System.Net.Http.Http2Connection.Http2Stream.SendRequestBodyAsync(CancellationToken cancellationToken)
   at System.Net.Http.Http2Connection.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, CancellationTokenSource cts, Boolean disposeCts, Int64 timeoutTime, CancellationToken originalCancellationToken)
   at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter[T](Task`1 task, TimeSpan timeout, String filePath, Int32 lineNumber) in /_/src/Testing/src/TaskExtensions.cs:line 31
   at Microsoft.AspNetCore.Testing.TaskExtensions.TimeoutAfter[T](Task`1 task, TimeSpan timeout, String filePath, Int32 lineNumber) in /_/src/Testing/src/TaskExtensions.cs:line 31
   at Interop.FunctionalTests.HttpClientHttp2InteropTests.Settings_MaxConcurrentStreamsPost_Server(String scheme) in /_/src/Servers/Kestrel/test/Interop.FunctionalTests/HttpClientHttp2InteropTests.cs:line 1296
--- End of stack trace from previous location ---
```

**Logs:**
<details>

```
[0.001s] [TestLifetime] [Information] Starting test Settings_MaxConcurrentStreamsPost_Server-http at 2020-10-23T18:58:47
[0.005s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting starting
[0.012s] [Microsoft.Hosting.Lifetime] [Information] Now listening on: http://127.0.0.1:38933
[0.012s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Debug] Loaded hosting startup assembly Interop.FunctionalTests
[0.012s] [Microsoft.Hosting.Lifetime] [Information] Application started. Press Ctrl+C to shut down.
[0.012s] [Microsoft.Hosting.Lifetime] [Information] Hosting environment: Production
[0.012s] [Microsoft.Hosting.Lifetime] [Information] Content root path: /home/helixbot/work/AF23099D/w/A1E90916/e/
[0.012s] [Microsoft.Extensions.Hosting.Internal.Host] [Debug] Hosting started
[0.013s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D" accepted.
[0.014s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D" started.
[0.014s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending SETTINGS frame for stream ID 0 with length 18 and flags NONE
[0.014s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending WINDOW_UPDATE frame for stream ID 0 with length 4 and flags 0x0
[0.014s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received SETTINGS frame for stream ID 0 with length 6 and flags NONE
[0.014s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending SETTINGS frame for stream ID 0 with length 0 and flags ACK
[0.015s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received WINDOW_UPDATE frame for stream ID 0 with length 4 and flags 0x0
[0.016s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 1 with length 53 and flags END_HEADERS
[0.017s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 3 with length 53 and flags END_HEADERS
[0.017s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 5 with length 53 and flags END_HEADERS
[0.017s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 7 with length 53 and flags END_HEADERS
[0.017s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 9 with length 53 and flags END_HEADERS
[0.017s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.018s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D" reached the maximum number of concurrent HTTP/2 streams allowed.
[0.018s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 11 with length 53 and flags END_HEADERS
[0.018s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D": HTTP/2 stream error.
Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2StreamErrorException: HTTP/2 stream ID 11 error (REFUSED_STREAM): A new stream was refused because this connection has reached its stream limit.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.StartStream() in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 1037
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.DecodeHeadersAsync(Boolean endHeaders, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 973
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessHeadersFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 618
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 463
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessRequestsAsync[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 242
[0.019s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 11 with length 4 and flags 0x0
[0.021s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.021s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 13 with length 53 and flags END_HEADERS
[0.021s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D": HTTP/2 stream error.
Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2StreamErrorException: HTTP/2 stream ID 13 error (REFUSED_STREAM): A new stream was refused because this connection has reached its stream limit.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.StartStream() in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 1037
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.DecodeHeadersAsync(Boolean endHeaders, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 973
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessHeadersFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 618
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 463
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessRequestsAsync[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 242
[0.022s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 13 with length 4 and flags 0x0
[0.022s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 15 with length 53 and flags END_HEADERS
[0.022s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D": HTTP/2 stream error.
Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2StreamErrorException: HTTP/2 stream ID 15 error (REFUSED_STREAM): A new stream was refused because this connection has reached its stream limit.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.StartStream() in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 1037
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.DecodeHeadersAsync(Boolean endHeaders, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 973
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessHeadersFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 618
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 463
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessRequestsAsync[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 242
[0.022s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 15 with length 4 and flags 0x0
[0.023s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 17 with length 53 and flags END_HEADERS
[0.023s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D": HTTP/2 stream error.
Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2StreamErrorException: HTTP/2 stream ID 17 error (REFUSED_STREAM): A new stream was refused because this connection has reached its stream limit.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.StartStream() in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 1037
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.DecodeHeadersAsync(Boolean endHeaders, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 973
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessHeadersFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 618
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessFrameAsync[TContext](IHttpApplication`1 application, ReadOnlySequence`1& payload) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 463
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2Connection.ProcessRequestsAsync[TContext](IHttpApplication`1 application) in /_/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs:line 242
[0.024s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 17 with length 4 and flags 0x0
[0.024s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.024s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.024s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.024s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 7 with length 45 and flags END_STREAM, END_HEADERS
[0.024s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 5.5540ms
[0.024s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000007": started reading request body.
[0.025s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000007": done reading request body.
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Information] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000007": the application completed without reading the entire request body.
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 3 with length 7 and flags END_STREAM, END_HEADERS
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 7 with length 4 and flags 0x0
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 5 with length 7 and flags END_STREAM, END_HEADERS
[0.026s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 7.6458ms
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000005": started reading request body.
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000005": done reading request body.
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Information] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000005": the application completed without reading the entire request body.
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 5 with length 4 and flags 0x0
[0.026s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 9 with length 7 and flags END_STREAM, END_HEADERS
[0.026s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 8.0119ms
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000009": started reading request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 1 with length 7 and flags END_STREAM, END_HEADERS
[0.027s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 6.1434ms
[0.027s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 9.1951ms
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000003": started reading request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000003": done reading request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Information] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000003": the application completed without reading the entire request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 3 with length 4 and flags 0x0
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000009": done reading request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Information] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000009": the application completed without reading the entire request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 9 with length 4 and flags 0x0
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000001": started reading request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000001": done reading request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 19 with length 53 and flags END_HEADERS
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Information] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000001": the application completed without reading the entire request body.
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 1 with length 4 and flags 0x0
[0.027s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received SETTINGS frame for stream ID 0 with length 0 and flags ACK
[0.027s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.028s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 19 with length 7 and flags END_STREAM, END_HEADERS
[0.028s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 7 with length 11 and flags NONE
[0.028s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 0.1828ms
[0.028s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000013": started reading request body.
[0.028s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000013": done reading request body.
[0.028s] [Microsoft.AspNetCore.Server.Kestrel] [Information] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000013": the application completed without reading the entire request body.
[0.028s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending RST_STREAM frame for stream ID 19 with length 4 and flags 0x0
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 19 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 5 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 1 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 9 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 17 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 3 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 13 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 15 with length 11 and flags NONE
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 5 with length 0 and flags END_STREAM
[0.029s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 7 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 17 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 3 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 13 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 19 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 9 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 15 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 21 with length 53 and flags END_HEADERS
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 21 with length 11 and flags NONE
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 21 with length 0 and flags END_STREAM
[0.030s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 23 with length 53 and flags END_HEADERS
[0.031s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 23 with length 11 and flags NONE
[0.031s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.031s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 21 with length 7 and flags END_STREAM, END_HEADERS
[0.031s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 23 with length 0 and flags END_STREAM
[0.031s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 1 with length 0 and flags END_STREAM
[0.031s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.031s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 23 with length 7 and flags END_STREAM, END_HEADERS
[0.031s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 0.1958ms
[0.031s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000017": started reading request body.
[0.031s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000017": done reading request body.
[0.031s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 1.3898ms
[0.032s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000015": started reading request body.
[0.032s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000015": done reading request body.
[0.032s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received HEADERS frame for stream ID 25 with length 53 and flags END_HEADERS
[0.032s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 25 with length 11 and flags NONE
[0.032s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" received DATA frame for stream ID 25 with length 0 and flags END_STREAM
[0.032s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request starting HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11
[0.032s] [Microsoft.AspNetCore.Server.Kestrel] [Verbose] Connection id "0HM3NGMLDUF0D" sending HEADERS frame for stream ID 25 with length 7 and flags END_STREAM, END_HEADERS
[0.032s] [Microsoft.AspNetCore.Hosting.Diagnostics] [Information] Request finished HTTP/2 POST http://127.0.0.1:38933/ text/plain;+charset=utf-8 11 - 200 0 - 0.3950ms
[0.032s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000019": started reading request body.
[0.033s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D", Request id "0HM3NGMLDUF0D:00000019": done reading request body.
[0.040s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets] [Debug] Connection id "0HM3NGMLDUF0D" received FIN.
[0.040s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D" is closed. The last processed stream ID was 25.
[0.041s] [Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets] [Debug] Connection id "0HM3NGMLDUF0D" sending FIN because: "The client closed the connection."
[0.041s] [Microsoft.AspNetCore.Server.Kestrel] [Debug] Connection id "0HM3NGMLDUF0D" stopped.
[0.042s] [TestLifetime] [Information] Finished test Settings_MaxConcurrentStreamsPost_Server-http in 0.0417163s
```

</details>

**Comment by BrennanConroy** [comment]
Blocked by https://github.com/dotnet/runtime/issues/42472

TLDR; race in HttpClient that can prevent retry on RST_FRAME and instead throw TaskCanceledException.
